src/auth/storage.js

```diff
@@ -24,30 +24,35 @@ class MemoryStorage {
 }
 
 class Storage {
-  constructor(store) {
+  /**
+   * @param {Storage} store implementer of Storage interface
+   * @param {String} namespace appID to be used for namespacing
+  */
+  constructor(store, namespace) {
     this.store = store;
+    this.namespace = namespace;
   }
 
-  get(key) { return new Promise(resolve => resolve(this.store.getItem(key))); }
-  set(key, value) { return new Promise(resolve => resolve(this.store.setItem(key, value))); }
-  remove(key) { return new Promise(resolve => resolve(this.store.removeItem(key))); }
+  get(key) { return new Promise(resolve => resolve(this.store.getItem(`${this.namespace}.${key}`))); }
+  set(key, value) { return new Promise(resolve => resolve(this.store.setItem(`${this.namespace}.${key}`, value))); }
+  remove(key) { return new Promise(resolve => resolve(this.store.removeItem(`${this.namespace}.${key}`))); }
   clear() { return new Promise(resolve => resolve(this.store.clear())); }
```

Comment:
[q] Shouldn't clear just clear the specific namespace? Otherwise you cannot maintain multiple apps and log out of  one.

Reply:
Good point, though this will be trickier than expected. `.clear` doesn't take any parameters– we'll have to keep a set of stored keys separately, and then iterate through each one to clear them out, never actually calling `.clear`.

Reply:
Actually looks like we don't call this clear method. Should we just remove it?

Reply:
We call it in tests... it might not be a bad idea to have? In theory, we should be calling it on logout. I'll look into it.

Reply:
```
clear() {
    return this.storage.remove(authCommon.USER_AUTH_KEY).then(() =>
      this.storage.remove(authCommon.REFRESH_TOKEN_KEY)
    ).then(() =>
      this.clearImpersonation
    );
  }
```

That is the `clear` method in the `Auth` class. Seems like it's trying to preserve some amount of saved data, which seems counter intuitive. Should we just be calling `this.storage.clear()` here?

Reply:
The device id should not be cleared. This looks correct if you look at the way Android does it